resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -250,11 +250,31 @@ private[spark] object Config extends Logging {
       .stringConf
       .createOptional
 
+  // the definition of a label in DNS (RFC 1123).
+  private val dns1123LabelFmt = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+
+  private val podConfValidator = (s"^$dns1123LabelFmt(\\.$dns1123LabelFmt)*$$").r.pattern
+
+  // The possible longest executor name would be "$prefix-exec-${Long.MaxValue}"
+  private def isValidExecutorPodNamePrefix(prefix: String): Boolean = {
+    val reservedLen = Long.MaxValue.toString.length + 6
+    val validLength = prefix.length + reservedLen <= KUBERNETES_LABEL_MAX_LENGTH
+    validLength && podConfValidator.matcher(prefix).matches()
+  }
+
   val KUBERNETES_EXECUTOR_POD_NAME_PREFIX =
     ConfigBuilder("spark.kubernetes.executor.podNamePrefix")
-      .doc("Prefix to use in front of the executor pod names.")
+      .doc("Prefix to use in front of the executor pod names. It must conform the rules defined" +
+        " by the Kubernetes <a href=\"https://kubernetes.io/docs/concepts/overview/" +
```

Comment:
If you don't mind, let's put the space at the end of the previous line instead of starting with space like `" by`.

Reply:
OK